src/librustdoc/html/render/sidebar.rs

```diff
@@ -279,10 +281,43 @@ fn sidebar_assoc_items<'a>(
 ) {
     let did = it.item_id.expect_def_id();
     let cache = cx.cache();
+    let tcx = cx.tcx();
+    let mut v: Vec<&Impl> =
+        cache.impls.get(&did).map(Vec::as_slice).unwrap_or(&[]).iter().collect();
+    if let TypeAliasItem(ait) = &*it.kind &&
```

Comment:
Unless I missed something, it's the same code as the one in https://github.com/rust-lang/rust/pull/115201/files#diff-40a0eb025da61717b3b765ceb7fab21d91af3012360e90b9f46e15a4047946faR1145. Could you write a common function to prevent duplicating the code please?